File: portal/src/lib/sports/sportEmoji.ts
/** Emoji associé au sport d’un club (libellés ClubSports Flutter, clés normalisées). */
const SPORT_EMOJI_BY_KEY: Record<string, string> = {
  football: "⚽",
  basketball: "🏀",
  volleyball: "🏐",
  handball: "🤾",
  rugby: "🏉",
  tennis: "🎾",
  natation: "🏊",
  athletisme: "🏃",
  judo: "🥋",
  escrime: "🤺",
  aviron: "🚣",
  autre: "🏅",
};

/**
 * Retourne l’emoji du sport du club, ou un trophée générique si inconnu.
 */
export function sportEmoji(sport: string | null | undefined): string {
  const key = (sport ?? "")
    .toLowerCase()
    .trim()
    .normalize("NFD")
    .replace(/\p{M}/gu, "")
    .replace(/-/g, "");
  if (!key) return "🏅";
  return SPORT_EMOJI_BY_KEY[key] ?? "🏅";
}

/**
 * Libellé club avec emoji sport (ex. « 🏐 viro volley »).
 */
export function clubLabelWithSportEmoji(params: {
  name: string;
  sport: string | null | undefined;
}): string {
  const name = params.name.trim() || "Club";
  return `${sportEmoji(params.sport)} ${name}`;
}

File: lib/utils/sport_emoji.dart
/// Emoji sport aligné sur le portail (`sportEmoji.ts`) et [ClubSports].
String sportEmoji(String? sport) {
  final key = _normalizeSportKey(sport);
  if (key.isEmpty) return '🏅';
  return _sportEmojiByKey[key] ?? '🏅';
}

/// Libellé club avec emoji sport (ex. « 🏐 viro volley »).
String clubLabelWithSportEmoji({required String name, String? sport}) {
  final trimmed = name.trim().isEmpty ? 'Club' : name.trim();
  return '${sportEmoji(sport)} $trimmed';
}

String _normalizeSportKey(String? sport) {
  if (sport == null) return '';
  var key = sport.toLowerCase().trim();
  const accents = {
    'à': 'a',
    'â': 'a',
    'ä': 'a',
    'é': 'e',
    'è': 'e',
    'ê': 'e',
    'ë': 'e',
    'î': 'i',
    'ï': 'i',
    'ô': 'o',
    'ö': 'o',
    'ù': 'u',
    'û': 'u',
    'ü': 'u',
    'ç': 'c',
  };
  final buffer = StringBuffer();
  for (final rune in key.runes) {
    final char = String.fromCharCode(rune);
    buffer.write(accents[char] ?? char);
  }
  return buffer.toString().replaceAll('-', '');
}

const Map<String, String> _sportEmojiByKey = {
  'football': '⚽',
  'basketball': '🏀',
  'volleyball': '🏐',
  'handball': '🤾',
  'rugby': '🏉',
  'tennis': '🎾',
  'natation': '🏊',
  'athletisme': '🏃',
  'judo': '🥋',
  'escrime': '🤺',
  'aviron': '🚣',
  'autre': '🏅',
};
